File: server/Models/ServiceTicket.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Yes.Models;

public class ServiceTicket
{
    [Key]
    public string Id { get; set; } = string.Empty;

    [Required]
    public string Customer_id { get; set; } = string.Empty;
    [ForeignKey("Customer_id")]
    public virtual Customer Customer { get; set; } = null!;
    [Required]
    public string Room_id { get; set; } = string.Empty;
    [ForeignKey("Room_id")]
    public virtual Room Room { get; set; } = null!;
    [Required]
    public string Service_id { get; set; } = string.Empty;
    [ForeignKey("Service_id")]
    public virtual Service Service { get; set; } = null!;

    [Required]
    [MaxLength(255)]
    public string Details { get; set; } = string.Empty;
    [Required]
    public int Status { get; set; } = 0;
}

File: frontend/types/ServiceTicket.ts
export interface ServiceTicket {
  id: string;
  customer_id: string;
  room_id: string;
  service_id: string;
  details: string;
  status: number;
  created_at: string;
  updated_at: string;
}

export interface CreateServiceTicketPayload {
  customer_id: string;
  room_id: string;
  service_id: string;
  details: string;
}

export interface UpdateServiceTicketPayload {
  customer_id?: string | null;
  room_id?: string | null;
  service_id?: string | null;
  details?: string | null;
}
